addons/stock_account/models/product.py

```diff
@@ -309,19 +309,22 @@ def _run_fifo_vacuum(self, company=None):
 
         :param company: recordset of `res.company` to limit the execution of the vacuum
         """
-        self.ensure_one()
         if company is None:
             company = self.env.company
+        done_products = self.browse()
         svls_to_vacuum = self.env['stock.valuation.layer'].sudo().search([
-            ('product_id', '=', self.id),
+            ('product_id', 'in', self.ids),
             ('remaining_qty', '<', 0),
             ('stock_move_id', '!=', False),
             ('company_id', '=', company.id),
         ], order='create_date, id')
         for svl_to_vacuum in svls_to_vacuum:
+            if svl_to_vacuum.product_id in done_products:
```

Comment:
do a single search to get all the fixing layers per layer to fix then sort them in python